hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/DNSDomainNameResolver.java

```diff
@@ -40,6 +48,16 @@ public String getHostnameByIP(InetAddress address) {
         && host.charAt(host.length()-1) == '.') {
       host = host.substring(0, host.length()-1);
     }
+    // Protect against the Java behaviour of returning the IP address as a string from a cache
+    // instead of performing a reverse lookup.
+    if (host.equals(address.getHostAddress())) {
```

Comment:
should we check that the `host` is not null? I saw we check that at line 47.

Reply:
What's the expected behavior here? does the called expect us to throw or is it acceptable to return `null` ?

Reply:
In fact, it should never be null according to the docs on `getCanonicalHostName`.

Reply:
It either returns what it figured out, or returns the IP address as a string.  There is no way of knowing when it is failing to provide an actual FQDN, which is why there is the additional check to see if it is just the IP address.